### ISSUE TITLE
Issue migrating a user picker custom field

### ISSUE THREAD

**Issue by RoyalWulf** [solidify/jira-azuredevops-migrator#794]
**Describe the problem**
This is a similar issue to 
[https://github.com/solidify/jira-azuredevops-migrator/issues/179](url)
[https://github.com/solidify/jira-azuredevops-migrator/issues/191](url)

In Jira we have a user picker custom field named BA, but it is not exported until it has a second revision

I create a new user story and I give BA (user picker custom field) a value.
If I run jira-export it does not export the custom field.  
See US-27012 - V1.json.txt below for the user story json file

If I make a change to the field BA and run jira-export, it does export the field. 
See US-27012-V2.json.txt below.

have attached the export log for the two exports.

this should work the first time.

A clear and concise description of what the problem/bug is.

**To Reproduce**
run jira-export

**Tool version**
3.0.136

**Attachments**

Please attach the following files:
[US-27012 - V1.json.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11843182/US-27012.-.V1.json.txt)

[US-27012-V2.json.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11843256/US-27012-V2.json.txt)

[config-agile_SMS.json.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11843260/config-agile_SMS.json.txt)

[jira-export-log-230623-134237_V1.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11843261/jira-export-log-230623-134237_V1.txt)

[jira-export-log-230623-134708_V2.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11843263/jira-export-log-230623-134708_V2.txt)

**Screenshots**

**Comment by Alexander-Hjelm** [comment]
@RoyalWulf, I have successfully migrated a custom issue picker. Here is how I did it. Give it a try and let me know if your issue still persists:

Custom field `alexander-testar-custom-userpicker` on my Jira issue:

![image](https://github.com/solidify/jira-azuredevops-migrator/assets/10683896/bf43c743-bf2a-4f2a-ae1c-06d2b418ea30)

Inspecting the REST API response gives me the field key:

![image](https://github.com/solidify/jira-azuredevops-migrator/assets/10683896/d89da718-778c-41f8-859a-5b9330313ee9)

So I set up the following field map:

![image](https://github.com/solidify/jira-azuredevops-migrator/assets/10683896/3706d070-2d60-498e-96e4-bed71bd64f47)

And the resulting field becomes:

![image](https://github.com/solidify/jira-azuredevops-migrator/assets/10683896/1e4c8b18-1809-4111-9cff-492bfc371e44)

**Comment by RoyalWulf** [comment]
The issue I had was it worked if there had been a change to the value ie had history but didn't work if there was no change.
Will test but busy on other things at the moment

**Comment by Alexander-Hjelm** [comment]
After some digging I realized that this is exactly the edge case we solved and are now supporting in Jira Azure DevOps Migrator PRO. So unfortunately we will officially support custom userpicker fields in the PRO version and not in the Community version.

You can read more about the PRO offering here: https://github.com/solidify/jira-azuredevops-migrator#jira-azure-devops-migrator-pro

For some details, the field looks like this in the REST API response:

```json
  "customfield_10065": {
    "self": "https://solidifydemo.atlassian.net/rest/api/2/user?accountId=6038bfcc25b84e00696240d3",
    "accountId": "6038bfcc25b84e00696240d3",
    "emailAddress": "alexander.hjelm@solidify.dev",
    "avatarUrls": {
      "48x48": "https://secure.gravatar.com/avatar/e57bc21be69a2940bfa10a99ab4f6d74?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FAH-1.png",
      "24x24": "https://secure.gravatar.com/avatar/e57bc21be69a2940bfa10a99ab4f6d74?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FAH-1.png",
      "16x16": "https://secure.gravatar.com/avatar/e57bc21be69a2940bfa10a99ab4f6d74?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FAH-1.png",
      "32x32": "https://secure.gravatar.com/avatar/e57bc21be69a2940bfa10a99ab4f6d74?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FAH-1.png"
    },
    "displayName": "Alexander Hjelm",
    "active": true,
    "timeZone": "Europe/Berlin",
    "accountType": "atlassian"
  },
```

The PRO version lets you set up a **custom-key** property of your field mapping, which extracts e.g. the **emailAddress** or **accountId** property of the field, and is meant for cases where no properties such as **id** or **name** are present.

```json
      {
        "source": "customfield_10065",
        "custom-key": "emailAddress",
        "target": "Custom.CustomUserPicker",
        "mapper": "MapUser"
      }
```

**Comment by Alexander-Hjelm** [comment]
Merged #873 . Reopen the issue if the problem still persists.

**Comment by Alexander-Hjelm** [comment]
This was the field map that ended up working for me. `source` should be the field name:

```
      {
        "source": "Userpicker Field Name Jira",
        "target": "Custom.CustomUserPicker",
        "source-type": "name",
        "mapper": "MapUser"
      },
```
